plugins/uefi-dbx/fu-uefi-dbx-plugin.c

```diff
@@ -87,15 +87,17 @@ fu_uefi_dbx_plugin_constructed(GObject *obj)
 	FuPlugin *plugin = FU_PLUGIN(obj);
 	g_autoptr(FuVolume) esp = NULL;
 	g_autoptr(GError) error_udisks2 = NULL;
+	FuUefiDbxPlugin *self = FU_UEFI_DBX_PLUGIN(plugin);
+	FuContext *ctx = fu_plugin_get_context(plugin);
 
 	fu_plugin_add_rule(plugin, FU_PLUGIN_RULE_METADATA_SOURCE, "uefi_capsule");
 	fu_plugin_add_firmware_gtype(plugin, NULL, FU_TYPE_EFI_SIGNATURE_LIST);
 	fu_plugin_add_device_gtype(plugin, FU_TYPE_UEFI_DBX_DEVICE);
 
-	if (fu_snap_is_in_snap()) {
-		FuUefiDbxPlugin *self = FU_UEFI_DBX_PLUGIN(plugin);
+	/* only enable snapd integration if either running inside a snap or we detect that this is a
+	snapd FDE setup */
+	if (fu_snap_is_in_snap() || fu_context_has_flag(ctx, FU_CONTEXT_FLAG_FDE_SNAPD)) {
 		g_autoptr(GError) error_local = NULL;
-		/* only enable snapd integration if running inside a snap */
 		if (!fu_uefi_dbx_plugin_snapd_notify_init(FU_UEFI_DBX_PLUGIN(obj), &error_local)) {
 			/* specific error code if relevant APIs are not present and thus
 			 * integration cannot be supported */
```

Comment:
maybe the comment could be rephrased to make it more clear
```suggestion
			/* unless we got specific error code indicating lack of 
			 * relevant APIs, snapd integration is considered to be
			 * supported, even if snapd itself cannot be reached */
```